### PR TITLE
Forward text messages to operator when session in forwarding mode

### LINKED ISSUE #261: [human override] Bot in forwarding mode
Check for the status flag of #259 and forward all incoming messages to the UI using the message follow up id specified in the session, so they can be stored as part of the same chain

### PR DESCRIPTION
**Pull request by aeatencio**
Depends on instedd/aida-ui#231
Resolves #261

### CODE REVIEW

File: test/aida/bot_test.exs [hunk 1]
```diff
@@ -94,6 +94,80 @@ defmodule Aida.BotTest do
...
+                 HTTPoison.post(
+                   "#{bot.notifications_url}/messages/#{forward_messages_id}",
+                   %{type: "text", direction: "uto", content: "a message"}
+                   |> Poison.encode!()
+                 )
+               )
+      end
+    end
+
+    test "does not forward when session with blank .forward_messages_id", %{
```
Comment (nekron) [inline test/aida/bot_test.exs]: Maybe we could prevent this case altogether by filtering the forward messages id at the controller level instead of having to filter it on the bot. Either we have a valid `.forward_messages_id` or we don't have anything, no need to specifically test and escape the empty string case.

File: test/aida/bot_test.exs [hunk 2]
```diff
@@ -853,6 +927,31 @@ defmodule Aida.BotTest do
...
+        assert error_log.session_id == session.id
+
+        extra = %{
+          post_url: "#{bot.notifications_url}/messages/#{forward_messages_id}",
+          post_body: %{type: "text", direction: "uto", content: "a message"},
+          response: response
+        }
+
+        assert error_log.message ==
+                 "Error posting notification message to remote endpoint #{inspect(extra)}"
```
Comment (nekron) [inline test/aida/bot_test.exs]: I would prefer to avoid the `inspect()` in the error logs. This is supposed to be a human readable message, not an error dump. You could take a look at `Aida.ErrorHandler` which creates a Sentry log, then logs the error in the standard log, and then opens an error log with the human readable details.

Maybe we can add some method there to handle error logs for something else besides exceptions.

File: lib/aida/bot.ex [hunk 3]
```diff
@@ -156,7 +208,7 @@ defmodule Aida.Bot do
     post_notification(notifications_url, notification_type, message_data)
   end
 
-  defp log_incoming(%{bot: %{public_keys: public_keys} = bot} = message) do
+  defp log_incoming(%{bot: %{public_keys: public_keys} = bot} = message, direction \\ "incoming") do
```
[the remark this answers is not in the document]
Comment (aeatencio) [reply lib/aida/bot.ex]: Totally! This is a remnant of a discarded approach. Thanks!

File: test/aida/bot_test.exs (hunk 1 above)
Comment (aeatencio) [reply test/aida/bot_test.exs]: This test was written to support an actual case for the current instedd/aida-ui#231. Maybe we should prevent this case at the controller level, but I don't see any bad side effect on leaving this test (and its corresponding guard in bot) as a safeguard. What do you think?

File: test/aida/bot_test.exs (hunk 2 above)
Comment (aeatencio) [reply test/aida/bot_test.exs]: I think I've got it. I'll add some method to `Aida.ErrorHandler`, using `Logger.error` for the standard error and `ErrorLog.write` for a human readable one. Please let me know if something doesn't feel right. Thanks!

File: test/aida/bot_test.exs (hunk 1 above)
Comment (nekron) [reply test/aida/bot_test.exs]: I think that if we start doing this kind of things, in short we will be checking everything in the session and that's far from ideal.

I think it's better to ensure that the session is consistent.

It can be done at the session level. Maybe we can ensure that `Session.put` won't store any invalid value and clear the key instead, the same way we are already doing it with `nil` to avoid keys without a value.
